### ISSUE TITLE
gem update legacy behavior

### ISSUE THREAD

**Issue by hurricup** [rubygems/rubygems#6197]
Until some point rubygems were pretty dumb and attempted to install incompatible version of the gem if one was restricted by ruby version. But this was fixed in 3.2.3

However, `gem update` kept this "dumb" behavior.
E.g, here is what is going on when I use ruby 2.5.9:
```
hurricup@AM-UNIT-0002:~$ gem install rubygems-update -v '~>3.3' --no-document && \
  update_rubygems --no-document && \
  gem uninstall rubygems-update -ax && \
  gem update --system --no-document

Successfully installed rubygems-update-3.3.26
1 gem installed
  Successfully built RubyGem
  Name: bundler
  Version: 2.3.26
  File: bundler-2.3.26.gem
Bundler 2.3.26 installed
RubyGems 3.3.26 installed
Regenerating binstubs
Regenerating plugins

------------------------------------------------------------------------------

RubyGems installed the following executables:
	/usr/share/rvm/rubies/ruby-2.5.9/bin/gem
	/usr/share/rvm/rubies/ruby-2.5.9/bin/bundle
	/usr/share/rvm/rubies/ruby-2.5.9/bin/bundler

Removing update_rubygems
Successfully uninstalled rubygems-update-3.3.26
Updating rubygems-update
ERROR:  Error installing rubygems-update:
	rubygems-update-3.4.1 requires Ruby version >= 2.6.0. The current ruby version is 2.5.9.
```
When i'm trying to run `gem update --system` with `rubygems` 3.3.26, it fails with requirement error.
But, if i run following, it works perfectly:
```
hurricup@AM-UNIT-0002:~$ gem install rubygems-update
Fetching rubygems-update-3.3.26.gem
Successfully installed rubygems-update-3.3.26
Parsing documentation for rubygems-update-3.3.26
Installing ri documentation for rubygems-update-3.3.26
Done installing documentation for rubygems-update after 97 seconds
1 gem installed
```

**Comment by deivid-rodriguez** [comment]
Nice catch @hurricup, thanks! Will look into fixing this.

**Comment by jeromedalbert** [comment]
The issue doesn't happen on Ruby versions >= 3.0, and Ruby versions below 3.1 are [end of life](https://www.ruby-lang.org/en/downloads/branches/) at the time of writing.

Since RubyGems only supports Rubies that are not end of life, should this issue be closed?

---

Steps to repro the issue on older Rubies:

1. Create this Dockerfile:

    ```Dockerfile
    FROM ruby:2.5

    # Fails with ruby:2.5, ruby:2.6, or ruby:2.7
    # Succeeds with ruby:3.0 and above
    RUN gem update --system
    ```

2. Run `docker build .`

**Comment by deivid-rodriguez** [comment]
This was fixed by https://github.com/rubygems/rubygems/pull/7334, released as RubyGems 3.5.4. Closing!
